### DISCUSSION TITLE
Error headless mode due to use of .document by micromark

### DISCUSSION THREAD

**Discussion by terpimost** [remarkjs/remark#1037] [Q&A] [answered]
I think from version 9 to version 10 something changed with `remark-parse` and now it is Errors in the runtime in a worker thread.
I see that `mdast-util-from-markdown` uses  Micromark, which is using `decodeNamedCharacterReference` [here](https://github.com/micromark/micromark/blob/main/packages/micromark-core-commonmark/dev/lib/character-reference.js) which calls `document.createElement('i')` [here](https://github.com/wooorm/decode-named-character-reference/blob/main/index.dom.js) which throws a runtime error in a worker (headless usage) since there is no `document.` object in a worker

This is an actual call stack I have 
![image](https://user-images.githubusercontent.com/1903309/187304929-120c59e7-fd3d-41b5-9cf6-1818cf16e616.png)

Do you think there is any workaround?

**Comment by ChristianMurphy** [comment, marked as answer]
This is a bug in your configuration or build tool.
`decode-character-reference` provides both a web compatible and a worker compatible implementation.
https://github.com/wooorm/decode-named-character-reference/blob/ddb8cdf65a7ce4795c4fc45c8433495377b0f5ea/package.json#L32-L39
Your build tool chose the `browser` one rather than `worker`
